java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -44,15 +44,15 @@
 import java.util.stream.Collectors;
 
 public class DefaultJavaFileScannerContext implements JavaFileScannerContext {
-  private final CompilationUnitTree tree;
+  private final ThreadLocal<CompilationUnitTree> tree;
   @VisibleForTesting
-  private final SemanticModel semanticModel;
+  private final ThreadLocal<SemanticModel> semanticModel;
+  private final ThreadLocal<ComplexityVisitor> complexityVisitor;
+  private final ThreadLocal<File> file;
+  private final ThreadLocal<Boolean> fileParsed;
   private final SonarComponents sonarComponents;
-  private final ComplexityVisitor complexityVisitor;
-  private final File file;
-  private final JavaVersion javaVersion;
-  private final boolean fileParsed;
-
+  private JavaVersion javaVersion;
+/*
   public DefaultJavaFileScannerContext(CompilationUnitTree tree, File file, SemanticModel semanticModel,
                                        @Nullable SonarComponents sonarComponents, JavaVersion javaVersion, boolean fileParsed) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)